### ISSUE TITLE
How to reset gradient before each batch

### ISSUE THREAD

**Issue by lokxii** [huggingface/candle#2986]
In Pytorch, you would call `optimizer.zero_grad` to zero the gradients before every batch. How do you do this in candle?
